packages/drivers/odsp-driver/src/odspDocumentStorageManager.ts

```diff
@@ -658,13 +628,17 @@ export class OdspDocumentStorageService implements IDocumentStorageService {
                 const odspSnapshot = await fetchLatestSnapshotCore(
                     this.odspResolvedUrl,
                     this.getStorageToken,
-                    tokenFetchOptions,
                     snapshotOptionsWithoutBlobs,
                     this.logger,
                     snapshotDownloader,
                     putInCache);
                 return odspSnapshot;
             }
+            // Clear the cache on 401/403/404 on snapshot fetch from network because this means either the user doesn't have
+            // permissions for the file or it was deleted. So the user will again try to fetch from cache on any failure in future.
```

Comment:
maybe clarify it:
"So, if we do not clear cache, we will continue fetching snapshot from cache in the future."